### DISCUSSION TITLE
Set OTA password function

### DISCUSSION THREAD

**Discussion by denizsahin7744** [HomeSpan/HomeSpan#836] [Ideas]
Hi can you add a set ota password function to the library. I will give the code (for 1.9.0) to down

`Span& setOTAPassword(const char *pwd){spanOTA.setPassword(pwd);return(*this);}`
you need to add this code to line 325 of [https://github.com/HomeSpan/HomeSpan/blob/release-1.9.0/src/HomeSpan.h](url)

**Comment by HomeSpan** [comment]
I'm not sure I understand the request - you can already set the OTA password when you enable OTA using `homeSpan.enableOTA(const char *pwd)` (see the [OTA page](OTA.md)).

**Reply by denizsahin7744** [reply]
I know i can select the password of over the air when i want to enable it but i want to change the password of over the air without enabling it.

**Reply by HomeSpan** [reply]
Why?

**Reply by denizsahin7744** [reply]
I'm trying to create a something admin/developer panel. In the panel i can make changes like enabling weblog, selecting Ip adress, changing device name/access point name and more. I wanted this function included in homespan because everything i made didn't changes the source of homeSpan until the changing password of over the air.

**Reply by HomeSpan** [reply]
Yes, but shouldn't your developer panel allow for enabling/disabling the OTA.  And when it does, that's when it should pass a user-defined password.

**Reply by denizsahin7744** [reply]
I'm also working on a which changes password every 10 minute. When user want to login he needs to get a login key from external web server. When he entered the password esp32 visits the verification server with it's own serial number and verifies the login key.

**Reply by HomeSpan** [reply]
Note that changing the OTA password has no effect until the next re-start of the ESP32.

**Reply by denizsahin7744** [reply]
thanks i didn't know that but i have already added reboot after saving the changes.
